### PR TITLE
Add changelog for the v1.4.0-beta.1 release

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

Add the changelog for the v1.4.0-beta.1 release.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @kron4eg

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold
until #1734 is not merged

**Review by kron4eg — APPROVED** [review]

**Pull request merged**
